### ISSUE TITLE
DOC: In search for the matplotlib codex

### ISSUE THREAD

**Issue by ImportanceOfBeingErnest** [matplotlib/matplotlib#11151]
### The codex

The [matplotlib search](https://matplotlib.org/search.html?q=codex) tells us about the "codex":

> If you want to limit your search to working code examples, include the keyword "codex" (mnemonic for code example) in your search, e.g., "codex ellipse"; see search examples.

and the [FAQ: Search examples](https://matplotlib.org/faq/howto_faq.html#search-examples)

>The nearly 300 code Gallery included with the Matplotlib source distribution are full-text searchable from the Search Page page, but sometimes when you search, you get a lot of results from the The Matplotlib API or other documentation that you may not be interested in if you just want to find a complete, free-standing, working piece of example code. To facilitate example searches, we have tagged every code example page with the keyword codex for code example which shouldn’t appear anywhere else on this site except in the FAQ. So if you want to search for an example that uses an ellipse, Search Page for codex ellipse.

### The problem

Yet, the codex has disappeared from the gallery - probably due to the migration to sphinx-gallery.

In [older versions](https://matplotlib.org/2.0.2/examples/pylab_examples/bar_stacked.html), this looked like

![image](https://user-images.githubusercontent.com/23121882/39457345-9cb1bc76-4ced-11e8-8867-c5726a733578.png)

### Possible solutions

1. Remove all references to "codex", as it's more than confusing to be told to look for something that doesn't exist.
2. Revive the codex by
    * manually adding it to all examples
    * check in how far some kind of template can be used for this.

The latter would be related to 
* Examples not shown in API docs for many methods. #10974
* DOC: add links to examples for a few examples #11019

**Comment by tacaswell** [comment]
This seems like something we should be able to inject via sphinx gallery...

**Comment by ImportanceOfBeingErnest** [comment]
Monkey patching would look as follows. Inside of `conf.py` you'd add something like this:

```
from sphinx_gallery import gen_rst
gen_rst.SPHX_GLR_SIG = """\n
.. only:: html

 .. rst-class:: sphx-glr-signature

    Keywords: Matplotlib code example, Codex\n"""
```

Is this the way to go here?

**Comment by ImportanceOfBeingErnest** [comment]
Since I didn't get any answer here I implemented the monkey-patching solution in #11169.
